src/testing/slim_check/testable.lean

```diff
@@ -581,18 +585,23 @@ exact $ add_decorations p
 end tactic
 
 /-- Run a test suite for `p` and return true or false: should we believe that `p` holds? -/
-def testable.check (p : Prop) (cfg : slim_check_cfg := {}) (p' : tactic.decorations_of p . tactic.mk_decorations) [testable p'] : io bool := do
-x ← io.run_rand (testable.run_suite p' cfg),
+def testable.check (p : Prop) (cfg : slim_check_cfg := {}) (p' : tactic.decorations_of p . tactic.mk_decorations) [testable p'] : io punit := do
+x ← match cfg.random_seed with
+    | some seed := io.run_rand_with seed (testable.run_suite p' cfg)
+    | none := io.run_rand (testable.run_suite p' cfg)
+    end,
 match x with
-| (success _) := io.put_str_ln ("Success") >> return tt
-| (gave_up n) := io.put_str_ln ("Gave up " ++ repr n ++ " times") >> return ff
+| (success _) := when cfg.random_seed.is_none $ io.put_str_ln "Success"
```

Comment:
What's going on here? Could you add some explanation of the `when` condition? It seems weird.

Reply:
Because the `random_seed` is used in test cases, this assumes that we want the test case to be quiet so it turns off the reporting of success. I could add a `quiet` flag if you think they're better separated or just makes clearer code.

Reply:
Yes, let's have `quiet`. I couldn't have worked out this intention from the current code.